File: 338-counting-bits/counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        arr = [0 for val in range(n+1)]
        power = -1
        target = 0
        for index in range(len(arr)):
            if index == 0:
                continue
            if index == 2 ** target:
                power+=1
                target+=1
            arr[index] = arr[index - 2 ** power] + 1
            
             
        
        return arr
            

File: 338-counting-bits/counting-bits.swift
class Solution {
    func countBits(_ n: Int) -> [Int] {
        var store : [Int] = Array(repeating:0, count:n+1);
        var r: Int = 0;
        var helper: Int = 0;
        var x: Int = 2;
        while(r <= n){
            if r == 0{
                r+=1;
                helper += 1;
                continue;
            }
            if r == x{
                helper = x;
                x = x * 2;
            }
            store[r] = 1 + store[r - helper];
            r+=1;
        }
        return store;
    }
}